src/univariate/discrete/geometric.jl

```diff
@@ -67,6 +67,13 @@ kurtosis(d::Geometric) = 6 + abs2(d.p) / (1 - d.p)
 
 entropy(d::Geometric) = (-xlogx(succprob(d)) - xlogx(failprob(d))) / d.p
 
+function kldivergence(p::Geometric, q::Geometric)
+    x = succprob(p)
+    y = succprob(q)
+    x == y && return zero(x)
+    return log(x) - log(y) + (inv(x) - one(x)) * (log1p(-x) - log1p(-y))
```

Comment:
I think there are some problems here:
- The function is not type stable, e.g. if the parameters are of different type and/or not floating point numbers
- The function would not work correctly if `x == 1` and `y != 1` - Distributions does not support it but actually it's a valid parameter for the Geometric distribution (it collapses) and hence it makes me wonder if there is a better implementation.

Maybe one could also work with the logits to combine the `log` and `log1p` calculations?

Maybe something like:
```suggestion
    succ_p = succprob(p)
    succ_q = succprob(q)
    if succ_p == succ_q || isone(succ_p)
        # Ensure that succ_p == succ_q is handled accurately
        return zero(float(succ_p / succ_q))
    elseif isone(succ_p)
        # Ensure that succ_p == 1 && succ_q < 1 is handled correctly
        return -log(succ_q / succ_p)
    else
        diff_logits = logit(succ_p) - logit(succ_q)
        return ((succ_p - 1) * diff_logits + log(succ_p / succ_q)) / succ_p
    end
```

Reply:
Does this pass the tests? I can't test locally at the moment.